internal/ws/client.go

```
@@ -49,27 +58,39 @@ type Client struct {
 func (c *Client) Serve() {
-	var heartBeatTicker = time.NewTicker(2 * time.Second)
+	// Write deadlines below: a peer that stops reading would otherwise block this goroutine and hold the socket open indefinitely.
+	var heartBeatTicker = time.NewTicker(pingPeriod)
 	defer heartBeatTicker.Stop()
 	defer c.Conn.Close()
-	
+
 	for {
 		select {
 		case <-heartBeatTicker.C:
+			c.Conn.SetWriteDeadline(time.Now().Add(writeWait))
```

golangci-lint (2.12.2)

[error] Error return value of `c.Conn.SetWriteDeadline` is not checked

(errcheck)